scrapy/http/request/__init__.py

```diff
@@ -40,8 +40,15 @@ def __init__(self, url, callback=None, method='GET', headers=None, body=None,
         self.dont_filter = dont_filter
 
         self._meta = dict(meta) if meta else None
+        self._kwargs = dict(kwargs) if kwargs else None
         self.flags = [] if flags is None else list(flags)
 
+    @property
+    def kwargs(self):
+        if self._kwargs is None:
```

Comment:
This check could go in `__init__`, right?
That way we can avoid this `if`

Reply:
I have no strong argument in favor of this pattern, just that it's the same way the `meta` attribute is handled.